app/models/source_parameter.rb

```diff
@@ -23,4 +23,12 @@ class SourceParameter < ApplicationRecord
   validates_presence_of :vita_partner_id
   validates_presence_of :code
   validates_uniqueness_of :code
+
+  before_validation :downcase_code
+
+  private
+
+  def downcase_code
+    self.code = code.downcase if code_changed?
```

Comment:
why is this conditional on a change? Are there scenarios when you don't want it to downcase?

Reply:
code_changed? is part of active model dirty -- it will only reassign the downcased code programmatically if the code is one of the values that changed. This prevents us from rewriting the downcased version of code on every write (for example, say we wanted to increment a "count" attribute every time we see a source param used -- we don't want to re-downcase the code every time we do that)